nltk/stem/paicehusk.py

```diff
@@ -0,0 +1,274 @@
+    ylp0.     { protect  -ply }
+    yl2>      { -ly > -    }
+    ygo1.     { -ogy > -og }
+    yhp1.     { -phy > -ph }
+    ymo1.     { -omy > -om }
+    ypo1.     { -opy > -op }
+    yti3>     { -ity > -   }
+    yte3>     { -ety > -   }
+    ytl2.     { -lty > -l  }
+    yrtsi5.   { -istry > - }
+    yra3>     { -ary > -   }
+    yro3>     { -ory > -   }
+    yfi3.     { -ify > -   }
+    ycn2t>    { -ncy > -nt }
+    yca3>     { -acy > -   }
+    zi2>      { -iz > -    }
+    zy1s.     { -yz > -ys  }
+    """
+
+    def __init__(self, ruletable=None):
+        """
+        :param ruletable: a string containing the rule data, separated
+            by newlines.
+        """
+        self.rules = defaultdict(list)
+        if ruletable:
+            self.read_rules(ruletable)
+        else:
+            self.read_rules(self.defaultrules)
+
+    def read_rules(self, ruletable):
+        """Verify integrity of ruletable.
+        """
+        rule_expr = self.rule_expr
+        rules = self.rules
+
+        for line in ruletable.split("\n"):
+            line = line.strip()
+            if not line:
+                continue
+
+            match = rule_expr.match(line)
+            if match:
+                ending = match.group("ending")[::-1]
+                lastchar = ending[-1]
+                intact = match.group("intact") == "*"
+                num = int(match.group("num"))
+                append = match.group("append")
+                cont = match.group("cont") == ">"
+
+                rules[lastchar].append((ending, intact, num, append, cont))
+            else:
+                raise Exception("Bad rule: %r" % line)
+
+    def first_vowel(self, word):
+        """Returns index of vowel or y.
+        """
+        vp = min([p for p in [word.find(v) for v in "aeiou"]
+                  if p > -1])
+        yp = word.find("y")
+        if yp > 0 and yp < vp:
+            return yp
+        return vp
+
+    def strip_prefix(self, word):
+        """Remove prefix from a word.
+        """
+        for prefix in ("kilo", "micro", "milli", "intra", "ultra", "mega",
+                       "nano", "pico", "pseudo"):
+            if word.startswith(prefix):
+                return word[len(prefix):]
+        return word
+
+    def stem(self, word):
+        """Returns a stemmed version of the argument string.
+        """
+
+        rules = self.rules
+        match = self.stem_expr.match(word)
+        if not match:
+            return word
+        stem = self.strip_prefix(match.group(0))
+
+        is_intact = True
+        continuing = True
+        while continuing:
+            pfv = self.first_vowel(stem)
+            rulelist = rules.get(stem[-1])
+            if not rulelist:
+                break
+
+            continuing = False
+            for ending, intact, num, append, cont in rulelist:
+                if stem.endswith(ending):
+                    if intact and not is_intact:
+                        continue
```

Comment:
This if-condition `if intac and not is_intact` seems a little funny. Maybe some comments on this would help users and us better maintain the code